client/src/index.jsx

```diff
@@ -2,21 +2,44 @@ import React from 'react';
 import ReactDom from 'react-dom';
 import SearchBar from './components/SearchBar.jsx'
 import Reservations from './components/Reservations.jsx'
+
 class App extends React.Component {
     constructor(props) {
         super(props);
+        this.state = {
+            hotelRooms: { rooms: [] },
+            startDate: 0,
+            endDate: 0
+        }
+    }
 
+    componentDidMount() {
+        this.initializeRoom()
     }
+    
+    initializeRoom() {
+        fetch('http://localhost:3003/api')
+        .then(response => response.json())
+        .then(response => {
+            console.log(response)
+            // this.setState(prevState => ({
+            //     hotelRooms: [...prevState, response.rooms]
+            // }))
+            this.setState({hotelRooms: response})
+            console.log(this.state.hotelRooms)

```

Comment:
Were you trying to see the UPDATED state after calling setState? If so, put a callback as 2nd argument in setState function would help.
`this.setState({hotelRooms: response}, ()=> console.log(this.state.hotelRooms))`

Reply:
Great observation by datonedoe here. `setState` is async!